File: actor-bench/actor_bench.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func plus(a, b int) (int, int, int) {
	return a, b, a + b
}

type Actor struct {
	x  float32
	y  float32
	vx float32
	vy float32
}

func update(actor *Actor) {
	actor.x += actor.vx
	actor.y += actor.vy
}

func main() {
	if len(os.Args) <= 1 {
		return
	}

	num, err := strconv.Atoi(os.Args[1])

	if err != nil {
		return
	}

	start := time.Now()

	const LEN = 10000
	list := make([]*Actor, 10000)

	for i := 0; i < LEN; i++ {
		newActor := Actor{}
		newActor.x = float32(i) / 10.0
		newActor.y = (float32(i) * 2) / 10.0
		newActor.vx = float32(i) / 100.0
		newActor.vy = (float32(i) * 2) / 100.0

		list[i] = &newActor
	}

	for i := 0; i < num; i++ {
		for j := 0; j < LEN; j++ {
			update(list[j])
		}
	}

	fmt.Println(time.Since(start).Nanoseconds() / 10000)

	// fmt.Println(list[5000].x)
	// fmt.Println(list[5000].y)
}

File: actor-bench/actor_bench.rs
use std::time::Instant;
use std::env;
const LEN: usize = 10000;

fn main() {
  let args: Vec<String> = env::args().collect();
  if args.len() <= 1 {
    return;
  }

  let num = args[1].parse().unwrap();

  let now = Instant::now();

  let mut list : Vec< Actor> = Vec::new();

  for i in 0..LEN {
    let f = i as f64;
    list.push( Actor{x:f/10.0, y: f*2.0/10.0, vx:f/100.0, vy: f*2.0/100.0});
  }

  for _i in 0..num {
    for j in 0..LEN {
      let actor = &mut list[j];
      actor.update();
    }
  }

  let new_now = Instant::now();
    
  println!("{}", list[5000].x);
  println!("{}", list[5000].y);
  println!("{:?}", new_now.duration_since(now));  
}

struct Actor {
  pub x : f64,
  pub y : f64,
  pub vx : f64,
  pub vy : f64,
}

impl Actor {
  pub fn update(&mut self) {
    self.x += self.vx;
    self.y += self.vy;
  }
}
